DylibBinder/XmlGenerator.cs

```diff
@@ -0,0 +1,289 @@
+			}
+		}
+
+		void WriteTypeDeclarations (List <DBTypeDeclaration> typeDeclarations)
+		{
+			foreach (var typeDeclaration in typeDeclarations) {
+
+				if (InnerX.IsInnerType (typeDeclaration))
+					continue;
+
+				WriteTypeDeclarationOpening (typeDeclaration);
+
+				if (typeDeclaration.Kind == TypeKind.Protocol)
+					WriteProtocolTypeDeclaration (typeDeclaration);
+				else
+					WriteTypeDeclaration (typeDeclaration);
+				writer.WriteEndElement ();
+			}
+		}
+
+		void WriteTypeDeclarationOpening (DBTypeDeclaration typeDeclaration)
+		{
+			writer.WriteStartElement ("typedeclaration");
+			WriteAttributeStrings (("kind", typeDeclaration.Kind), ("name", typeDeclaration.Name),
+								  ("accessibility", typeDeclaration.Accessibility), ("isDeprecated", typeDeclaration.IsDeprecated),
+								  ("isUnavailable", typeDeclaration.IsUnavailable), ("isObjC", typeDeclaration.IsObjC),
+								  ("isFinal", typeDeclaration.IsFinal));
+		}
+
+		void WriteTypeDeclaration (DBTypeDeclaration typeDeclaration)
+		{
+			WriteMembers (typeDeclaration);
+			WriteInnerTypes (typeDeclaration);
+			WriteGenericParameters (typeDeclaration);
+			WriteAssocTypes (typeDeclaration);
+			WriteElements (typeDeclaration);
+		}
+
+		void WriteProtocolTypeDeclaration (DBTypeDeclaration typeDeclaration)
+		{
+			WriteProtocolMembers (typeDeclaration);
+		}
+
+		void WriteMembers (DBTypeDeclaration typeDeclaration)
+		{
+			writer.WriteStartElement ("members");
+			WriteFuncs (typeDeclaration);
+			WriteProperties (typeDeclaration);
+			writer.WriteEndElement ();
+		}
+
+		void WriteProtocolMembers (DBTypeDeclaration typeDeclaration)
+		{
+			writer.WriteStartElement ("members");
+			WriteFuncs (typeDeclaration);
+			writer.WriteEndElement ();
+		}
+
+		void WriteInnerTypes (DBTypeDeclaration typeDeclaration)
+		{
+			foreach (var innerType in typeDeclaration.InnerTypes.InnerTypeCollection) {
+				writer.WriteStartElement ($"inner{innerType.Kind}");
+				WriteTypeDeclarationOpening (innerType);
+				WriteTypeDeclaration (innerType);
+				foreach (var innerInnerType in innerType.InnerTypes.InnerTypeCollection) {
+					WriteInnerTypes (innerInnerType);
+				}
+				writer.WriteEndElement ();
+				writer.WriteEndElement ();
+			}
+		}
+
+		void WriteGenericParameters (DBTypeDeclaration typeDeclaration)
+		{
+			if (typeDeclaration.GenericParameters.GenericParameterCollection.Count == 0)
+				return;
+
+			writer.WriteStartElement ("GenericParameterCollection");
+			foreach (var gp in typeDeclaration.GenericParameters.GenericParameterCollection) {
+				WriteGenericParameter (gp);
+			}
+			writer.WriteEndElement ();
+		}
+
+		void WriteGenericParameters (DBGenericParameters GenericParameterCollection)
+		{
+			var validGenericParameters = from g in GenericParameterCollection.GenericParameterCollection
+										 where g.Depth > 0
+										 select g;
+
+			if (validGenericParameters.Count () == 0)
```

Comment:
`Any` is faster, since it won't have to count all elements, just find one (or none):

```suggestion
			if (!validGenericParameters.Any ())
```